modules/core/src/main/scala/schema.scala

```diff
@@ -23,12 +23,57 @@ trait Schema {
 
   def pos: SourcePos
 
-  /** The types defined by this `Schema`. */
-  def types: List[NamedType]
+  /** The types defined by this `Schema` prior to any extensions. */
+  def unExtendedTypes: List[NamedType]
+
+  /** The types defined by this `Schema` with any extensions applied. */
+  def types: List[NamedType] = {
+    unExtendedTypes.map { tpe =>
+      val exts = extensions.filter(_.extended.dealias =:= tpe)
```

Comment:
is this `dealias` necessary? I wasn't sure if `=:=` would take care of it or if another level of dealiasing is required?

Reply:
`=:=` on its own should be sufficient.